daemon/src/daemon.cpp

```diff
@@ -1302,41 +1339,207 @@ void Daemon::receivedCalibItems(const std::vector<CalibStruct>& newCalibs)
     }
 }
 
-void Daemon::onGpsPropertyUpdatedGeodeticPos(const GeodeticPos& pos)
+void Daemon::sendGeodeticPos(const GnssPosStruct& pos)
 {
     TcpMessage tcpMessage(TCP_MSG_KEY::MSG_GEO_POS);
-    (*tcpMessage.dStream) << pos.iTOW << pos.lon << pos.lat
-                          << pos.height << pos.hMSL << pos.hAcc
-                          << pos.vAcc;
+    (*tcpMessage.dStream) << pos;
     emit sendTcpMessage(tcpMessage);
+}
+
+void Daemon::sendPositionModel(const PositionModeConfig& pos)
+{
+    TcpMessage tcpMessage(TCP_MSG_KEY::MSG_POSITION_MODEL);
+    (*tcpMessage.dStream) << pos;
+    emit sendTcpMessage(tcpMessage);
+}
+
+void Daemon::onGpsPropertyUpdatedGeodeticPos(const GnssPosStruct& pos)
```

Comment:
This function is way too long and complex. Split it into smaller parts to make it readable and maintainable.